testsuite/MDAnalysisTests/parallelism/test_multiprocessing.py

```diff
@@ -0,0 +1,177 @@
+
+from MDAnalysisTests.datafiles import (
+    CRD,
+    PSF, DCD,
+    DMS,
+    DLP_CONFIG,
+    DLP_HISTORY,
+    INPCRD,
+    GMS_ASYMOPT,
+    GMS_SYMOPT,
+    GRO,
+    GSD,
+    GSD_long,
+    LAMMPSdata_mini,
+    LAMMPSDUMP,
+    mol2_molecules,
+    MMTF,
+    NCDF,
+    PDB, PDB_small, PDB_multiframe,
+    PDBQT_input,
+    PQR,
+    TRR,
+    TRJ,
+    TRZ,
+    TXYZ,
+    XTC,
+    XPDB_small,
+    XYZ_mini, XYZ, XYZ_bz2,
+)
+
+
+@pytest.fixture(params=[
+    (PSF, DCD),
+    (GRO, XTC),
+    (PDB_multiframe,),
+    (XYZ,),
+    (XYZ_bz2,),  # .bz2
+    (GMS_SYMOPT,),  # .gms
+    (GMS_ASYMOPT,),  # .gz
+    (GSD_long,),
+    (NCDF,),
+    (np.arange(150).reshape(5, 10, 3).astype(np.float64),),
+    (GRO, [GRO, GRO, GRO, GRO, GRO]),
+    (PDB, [PDB, PDB, PDB, PDB, PDB]),
+    (GRO, [XTC, XTC]),
+])
+def u(request):
+    if len(request.param) == 1:
+        f = request.param[0]
+        return mda.Universe(f)
+    else:
+        top, trj = request.param
+        return mda.Universe(top, trj)
+
+
+# Define target functions here
+# inside test functions doesn't work
+def cog(u, ag, frame_id):
+    u.trajectory[frame_id]
+
+    return ag.center_of_geometry()
+
+
+def test_multiprocess_COG(u):
+    ag = u.atoms[2:5]
+
+    ref = np.array([cog(u, ag, i)
+                    for i in range(3)])
+
+    p = multiprocessing.Pool(2)
+    res = np.array([p.apply(cog, args=(u, ag, i))
+                    for i in range(3)])
+    p.close()
+    assert_equal(ref, res)
+
+
+def getnames(u, ix):
+    # Check topology stuff works
+    return u.atoms[ix].name
+
+
+def test_universe_unpickle_in_new_process():
+    u = mda.Universe(GRO, XTC)
+    ref = [getnames(u, i)
+           for i in range(3)]
+
+    p = multiprocessing.Pool(2)
+    res = [p.apply(getnames, args=(u, i))
+           for i in range(3)]
+    p.close()
+
+    assert_equal(ref, res)
+
+
+@pytest.fixture(params=[
+    # formatname, filename
+    ('CRD', CRD, dict()),
+    ('DATA', LAMMPSdata_mini, dict(n_atoms=1)),
+    ('DCD', DCD, dict()),
+    ('DMS', DMS, dict()),
+    ('CONFIG', DLP_CONFIG, dict()),
+    ('HISTORY', DLP_HISTORY, dict()),
+    ('INPCRD', INPCRD, dict()),
+    ('LAMMPSDUMP', LAMMPSDUMP, dict()),
+    ('GMS', GMS_ASYMOPT, dict()),
+    ('GRO', GRO, dict()),
+    ('GSD', GSD, dict()),
+    ('MMTF', MMTF, dict()),
+    ('MOL2', mol2_molecules, dict()),
+    ('PDB', PDB_small, dict()),
+    ('PQR', PQR, dict()),
+    ('PDBQT', PDBQT_input, dict()),
+    ('TRR', TRR, dict()),
+    ('TRZ', TRZ, dict(n_atoms=8184)),
+    ('TRJ', TRJ, dict(n_atoms=252)),
+    ('XTC', XTC, dict()),
+    ('XPDB', XPDB_small, dict()),
+    ('XYZ', XYZ_mini, dict()),
+    ('NCDF', NCDF, dict()),
+    ('TXYZ', TXYZ, dict()),
+    ('memory', np.arange(60).reshape(2, 10, 3).astype(np.float64), dict()),
+    ('CHAIN', [GRO, GRO, GRO], dict()),
+    ('CHAIN', [PDB, PDB, PDB], dict()),
+    ('CHAIN', [XTC, XTC, XTC], dict()),
+])
+def ref_reader(request):
```

Comment:
Would it make sense to put that test in `BaseReaderTest` (and maybe `_SingleFrameReader`) so any new reader get tested? While it looks cleaner to have the test here, I am worried we will miss readers, especially when new readers get added.

Reply:
Do you mean cover all the `_SingleFrameReader` as how `class TestPDBReader(_SingleFrameReader)` is implemented? As for pickle test in `BaseReaderTest` and `MutliframeReaderTest`, they were already added.

Reply:
It is what I meant, indeed. My concern when I see this fixture is that we will forget to add the new readers to the list. Ideally I would like something that fails if a reader is not tested for pickling and multiprocessing.

Reply:
I think for now, only tests in `NAMDBIN`, `PDB`, `PQR` utilize `_SingleFrameReader`. Makes sense to create a new PR for other single frame readers e.g. MMTF, to be inherited from that? (which it along can be backported.)

Reply:
I'd be in favor of getting the PR merged and open a new one where we think hard about how to do consistent testing of _all_ our readers and parsers. 

We'll also need to look at converters... but that's a different issue. Literally.